### PR TITLE
Misc Tycho fixes.

### PR DESCRIPTION
**Pull request by andyHa**
* Simplifies confirm-links and provides a deleteButton and dropdownDele…
* BREAKING: Marks all JavaScript marker classes with the "-js" suffix.
* Repairs input group appends for textfield.
* Permits to use tag and dot in right aligned table cells.
* Softens text-black a bit to match bootstrap colors.
* Provides a helper to reset the font-size to 1rem.
* Permits to mark btn-link as btn-danger.
* Cleanup of card scss.
* Prevents upscaling of card images.
* Permits to entirely disable the search field in a searchHeader.
* Properly names a controller and adds support for common barcode types.

### CODE REVIEW

File: src/main/java/sirius/web/util/BarcodeController.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    private void barcode(WebContext ctx, BarcodeFormat format) throws WriterException, IOException {
+        int width = ctx.getFirstFilled("w", "width").asInt(200);
+        int height = ctx.getFirstFilled("h", "height").asInt(200);
+        String content = ctx.getFirstFilled("c", "content").asString();
+        if (Strings.isEmpty(content)) {
+            ctx.respondWith().direct(HttpResponseStatus.BAD_REQUEST, "Usage: /barcode?type=qr&content=...&w=200&h=200");
+            return;
+        }
+
+        String fileType = ctx.getFirstFilled("fileType").asString("jpg");
```
Comment (meg1502) [inline src/main/java/sirius/web/util/BarcodeController.java:66]: isn't png better as a default for these kind of graphics?

**Review by meg1502 — APPROVED** [review]

File: src/main/java/sirius/web/util/BarcodeController.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     * <p>
+     * The parameter <tt>content</tt> determines the contents of the qr code. The parameters <tt>with</tt> and
+     * <tt>height</tt> its dimensions.
+     * </p>
+     *
+     * @param ctx the current request
+     * @throws Exception in case an error occurred when generating the qr code
+     */
+    @Routed(value = "/qr", priority = 999)
+    public void qr(WebContext ctx) throws Exception {
```
Comment (sabieber) [inline src/main/java/sirius/web/util/BarcodeController.java:48]: Optional: rename `ctx` to `webContext` in this file

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/web/util/BarcodeController.java (hunk 2 above)
Comment (qw3ry) [reply src/main/java/sirius/web/util/BarcodeController.java:48]: :+1: would also be consistent with line 53 then

File: src/main/java/sirius/web/util/BarcodeController.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            return;
+        }
+
+        String fileType = ctx.getFirstFilled("fileType").asString("jpg");
+        Writer writer = determineWriter(format);
+        BitMatrix matrix = writer.encode(content, format, width, height);
+        try (OutputStream out = ctx.respondWith()
+                                   .infinitelyCached()
+                                   .outputStream(HttpResponseStatus.OK,
+                                                 MimeHelper.guessMimeType("barcode." + fileType))) {
```
Comment (qw3ry) [inline src/main/java/sirius/web/util/BarcodeController.java:72]: Might the cache mess up things when different parameters come into play? I assume it won't just want to double check

**Review by qw3ry — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/web/util/BarcodeController.java (hunk 3 above)
Comment (andyHa) [reply src/main/java/sirius/web/util/BarcodeController.java:72]: The querystring should be part of the cache key..

**Review by jmuscireum — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by meg1502 — APPROVED** [review]

File: src/main/resources/default/taglib/t/dropdownDeleteItem.html.pasta [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<i:arg type="String" name="url"/>
+<i:arg type="Page" name="page" default=""/>
+<i:arg name="class" type="String" default="" description="Lists additional CSS classes to apply to the item."/>
+
+<i:pragma name="description" value="Renders a delete link to be used in a dropdown menu (e.g. card actions)."/>
+
+<t:dropdownItem class="danger confirm-link-js @class"
+            icon="fa fa-trash"
+            labelKey="NLS.delete"
+            url="@(page != null) ? page.linkToCurrentPage(url) : url" />
```
Comment (meg1502) [inline src/main/resources/default/taglib/t/dropdownDeleteItem.html.pasta:10]: parameter url is used but never used

**Pull request merged**